provider/provider.go

```diff
@@ -126,8 +172,20 @@ func (p *dockerNativeProvider) Diff(ctx context.Context, req *rpc.DiffRequest) (
 		}, nil
 	}
 
+	diff := map[string]*rpc.PropertyDiff{}
+	for key := range d.Adds {
+		diff[string(key)] = &rpc.PropertyDiff{Kind: rpc.PropertyDiff_ADD}
+	}
+	for key := range d.Deletes {
+		diff[string(key)] = &rpc.PropertyDiff{Kind: rpc.PropertyDiff_DELETE}
+	}
+	for key := range d.Updates {
+		diff[string(key)] = &rpc.PropertyDiff{Kind: rpc.PropertyDiff_UPDATE}
+	}
 	return &rpc.DiffResponse{
-		Changes: rpc.DiffResponse_DIFF_UNKNOWN,
+		Changes:         rpc.DiffResponse_DIFF_SOME,
```

Comment:
I believe you will want to conditionally set this, `if d.AnyChanges()` return `DiffResponse_DIFF_SOME`, otherwise return `DiffResponse_DIFF_NONE`.

Reply:
Looks like on L171 above we actually return DIFF_NONE to short-circuit so this current code looks right?

Reply:
Ah, `d == nil` is equivalent to `AnyChanges()`? 

Looks good to me then.